idp-service/src/main/java/io/mosip/idp/services/TokenServiceImpl.java

```diff
@@ -111,10 +125,11 @@ public void verifyClientAssertionToken(String clientId, String jwk, String clien
             throw new IdPException(ErrorConstants.INVALID_ASSERTION);
 
         try {
+      
             JWSKeySelector keySelector = new JWSVerificationKeySelector(JWSAlgorithm.RS256,
                     new ImmutableJWKSet(new JWKSet(RSAKey.parse(jwk))));
             JWTClaimsSetVerifier claimsSetVerifier = new DefaultJWTClaimsVerifier(new JWTClaimsSet.Builder()
-                    .audience(issuerId)
+                    .audience(Collections.singletonList((String)discoveryMap.get("token_endpoint")))
```

Comment:
I see we are setting a list here, why not string directly ?

Reply:
.audience is expecting List<String>